### ISSUE TITLE
[any test case] thread.c:2020:spdk_get_io_channel: *ERROR*: could not find io_device

### ISSUE THREAD

**Issue by shuhei-matsumoto** [spdk/spdk#1983]
We have seen many  failures as follows.

```
00:03:14.535   10:36:50	# tgt_rpc bdev_passthru_create -b Malloc3 -p PTBdevFromMalloc3
00:03:14.535   10:36:50	# /var/jenkins/workspace/crypto-autotest/spdk/scripts/rpc.py -s /var/tmp/spdk_tgt.sock bdev_passthru_create -b Malloc3 -p PTBdevFromMalloc3
00:03:14.668  [2021-06-07 10:36:50.431388] vbdev_passthru.c: 610:vbdev_passthru_register: *NOTICE*: Match on Malloc3
00:03:14.668  [2021-06-07 10:36:50.431420] vbdev_passthru.c: 638:vbdev_passthru_register: *NOTICE*: base bdev opened
00:03:14.668  [2021-06-07 10:36:50.431448] vbdev_passthru.c: 667:vbdev_passthru_register: *NOTICE*: io_device created at: 0x0x276e6b0
00:03:14.668  [2021-06-07 10:36:50.431454] vbdev_passthru.c: 682:vbdev_passthru_register: *NOTICE*: bdev claimed
00:03:14.668  [2021-06-07 10:36:50.431933] vbdev_passthru.c: 695:vbdev_passthru_register: *NOTICE*: pt_bdev registered
00:03:14.668  [2021-06-07 10:36:50.431947] vbdev_passthru.c: 696:vbdev_passthru_register: *NOTICE*: created pt_bdev for: PTBdevFromMalloc3
00:03:14.672  PTBdevFromMalloc3
00:03:14.680   10:36:50	# tgt_rpc bdev_null_create Null0 32 512
00:03:14.680   10:36:50	# /var/jenkins/workspace/crypto-autotest/spdk/scripts/rpc.py -s /var/tmp/spdk_tgt.sock bdev_null_create Null0 32 512
00:03:14.809  [2021-06-07 10:36:50.571884] thread.c:2020:spdk_get_io_channel: *ERROR*: could not find io_device 0x7f7481ff3480
00:03:14.809  [2021-06-07 10:36:50.571906] blobstore.c:3054:bs_channel_create: *ERROR*: Failed to create device channel.
00:03:14.809  [2021-06-07 10:36:50.571913] blobstore.c:5310:bs_register_md_thread: *ERROR*: Failed to get IO channel.
00:03:14.809  [2021-06-07 10:36:50.571930] thread.c:2020:spdk_get_io_channel: *ERROR*: could not find io_device 0x7f7481ff3480
00:03:14.809  [2021-06-07 10:36:50.571937] vbdev_gpt.c: 500:vbdev_gpt_read_gpt: *ERROR*: Failed to get an io_channel.
00:03:14.809  [2021-06-07 10:36:50.571948] vbdev_gpt.c: 553:vbdev_gpt_examine: *ERROR*: Failed to read info from bdev Null0
00:03:14.809  [2021-06-07 10:36:50.571958] thread.c:2020:spdk_get_io_channel: *ERROR*: could not find io_device 0x7f7481ff3480
00:03:16.089  Connection closed with partial response:
```

The recent commit https://github.com/spdk/spdk/commit/2246a9371809c30333b1844afbf9772c4b06db79 may be related.

## Link to the failed CI build

https://ci.spdk.io/results/autotest-per-patch/builds/49166/archive/crypto-autotest/build.log

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @mikeBashStuff. log: https://ci.spdk.io/public_build/autotest-per-patch_49213.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @mikeBashStuff. log: https://ci.spdk.io/public_build/autotest-per-patch_49216.html

**Comment by jimharris** [comment]
This is most likely due to recently merged commit 2246a9371809c30333b1844afbf9772c4b06.

I've pushed a revert for now: https://review.spdk.io/gerrit/c/spdk/spdk/+/8199 

Almost certainly the same as #1978.

**Comment by jimharris** [comment]
Patch merged.  Closing this issue.

We will keep #1977 open for now to track this general io_device lookup issue, should the revert not fix it.
